source/adapters/hip/memory.cpp

```diff
@@ -264,6 +264,9 @@ UR_APIEXPORT ur_result_t UR_APICALL urMemGetInfo(ur_mem_handle_t hMemory,
 
   switch (MemInfoType) {
   case UR_MEM_INFO_SIZE: {
+#if HIP_VERSION < 50600000
+    return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
```

Comment:
But it's only broken for `SurfaceMem`; `BufferMem` should still work. This code, however, returns the error for both.

Reply:
I think we should just `throw UR_RESULT_ERROR_UNSUPPORTED_FEATURE;` from `MemVisitor`, and then it will be caught and returned.